DQMOffline/L1Trigger/python/L1TEtSumJetOffline_cfi.py

```diff
@@ -41,11 +41,53 @@
 httEfficiencyBins.extend(list(xrange(400, 500, 50)))
 httEfficiencyBins.extend(list(xrange(500, 601, 10)))
 
+# from https://twiki.cern.ch/twiki/bin/view/CMS/JetID13TeVRun2017
+centralJetSelection = [
+    'abs(eta) <= 2.7',
+    'neutralHadronEnergyFraction < 0.9',
+    'neutralEmEnergyFraction < 0.9',
+    'numberOfDaughters > 1',
+    'muonEnergyFraction < 0.8'
+]
+withinTrackerSelection = centralJetSelection[:]
+withinTrackerSelection += [
+    'abs(eta) <= 2.4',
+    'chargedHadronEnergyFraction > 0',
+    'chargedMultiplicity > 0',
+    'chargedEmEnergyFraction < 0.8'
+]
+forwardJetSelection = [
+    'abs(eta) > 2.7',
+    'abs(eta) <= 3.0',
+    'neutralEmEnergyFraction > 0.02',
+    'neutralEmEnergyFraction < 0.99',
+    'neutralMultiplicity > 2'
+]
+veryForwardJetSelection = [
+    'abs(eta) > 3.0',
+    'neutralEmEnergyFraction < 0.99',
+    'neutralHadronEnergyFraction > 0.02',
+    'neutralMultiplicity > 10'
+]
+centralJetSelection = ' && '.join(centralJetSelection)
+withinTrackerSelection = ' && '.join(withinTrackerSelection)
+forwardJetSelection = ' && '.join(forwardJetSelection)
+veryForwardJetSelection = ' && '.join(veryForwardJetSelection)
+completeSelection = 'et > 30 && (' + ' || '.join([centralJetSelection, withinTrackerSelection,
+                                       forwardJetSelection, veryForwardJetSelection]) + ')'
+
+goodPFJetsForL1T = cms.EDFilter(
```

Comment:
Is this filter creating a new collection with selected PF jets (then it should rather be an EDProducer) or is it just filtering events based on the PF jet selection?
If this filter is filtering events based on the PF jet selection it needs to be in a separate path since otherwise the other offline analysis will be biased by the PF jet selected dataset.

I did not see `goodPFJetsForL1T` in any sequence except the testing one.

Reply:
`PFJetSelector` is an [existing EDFilter](https://github.com/cms-sw/cmssw/blob/09c3fce6626f70fd04223e7dacebf0b485f73f54/CommonTools/RecoAlgos/plugins/PFJetSelector.cc) that will produce a new collection.

`goodPFJetsForL1T` is added to the process when the DQM module is imported so CMSSW unscheduled processing should pick it up (that's what `consumes` should take care off). 
The only reason I added it to the test config is that I did not want to change to unscheduled mode (many changes).

Of course, this assumes that DQM nowadays runs in unscheduled mode (AFAIK all official modules should).

Reply:
You are right. Thanks for the clarification.